java/vector/src/main/codegen/templates/DenseUnionVector.java

```diff
@@ -0,0 +1,907 @@
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.util.TransferPair;
+
+<@pp.dropOutputFile />
+<@pp.changeOutputFile name="/org/apache/arrow/vector/complex/DenseUnionVector.java" />
+
+
+<#include "/@includes/license.ftl" />
+
+package org.apache.arrow.vector.complex;
+
+<#include "/@includes/vv_imports.ftl" />
+import io.netty.buffer.ArrowBuf;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import org.apache.arrow.memory.util.hash.ArrowBufHasher;
+import org.apache.arrow.memory.util.hash.SimpleHasher;
+import org.apache.arrow.vector.compare.VectorVisitor;
+import org.apache.arrow.vector.complex.impl.ComplexCopier;
+import org.apache.arrow.vector.util.CallBack;
+import org.apache.arrow.vector.ipc.message.ArrowFieldNode;
+import org.apache.arrow.memory.BaseAllocator;
+import org.apache.arrow.vector.BaseValueVector;
+import org.apache.arrow.vector.util.OversizedAllocationException;
+import org.apache.arrow.util.DataSizeRoundingUtil;
+import org.apache.arrow.util.Preconditions;
+
+import static org.apache.arrow.vector.types.UnionMode.Dense;
+
+
+
+/*
+ * This class is generated using freemarker and the ${.template_name} template.
+ */
+@SuppressWarnings("unused")
+
+
+/**
+ * A vector which can hold values of different types. It does so by using a StructVector which contains a vector for each
+ * primitive type that is stored. StructVector is used in order to take advantage of its serialization/deserialization methods,
+ * as well as the addOrGet method.
+ *
+ * For performance reasons, DenseUnionVector stores a cached reference to each subtype vector, to avoid having to do the struct lookup
+ * each time the vector is accessed.
+ * Source code generated using FreeMarker template ${.template_name}
+ */
+public class DenseUnionVector implements FieldVector {
+
+  private String name;
+  private BufferAllocator allocator;
+  int valueCount;
+
+  NonNullableStructVector internalStruct;
+  private ArrowBuf typeBuffer;
+  private ArrowBuf offsetBuffer;
+  private ArrowBuf validityBuffer;
+
+  /**
+   * The key is type Id, and the value is vector.
+   */
+  private ValueVector[] childVectors = new ValueVector[Byte.MAX_VALUE + 1];
+
+  /**
+   * The index is the type id, and the value is the type field.
+   */
+  private Field[] typeFields = new Field[Byte.MAX_VALUE + 1];
+
+  /**
+   * The next typd id to allocate.
+   */
+  private byte nextTypeId = 0;
+
+  private FieldReader reader;
+
+  private final CallBack callBack;
+  private long typeBufferAllocationSizeInBytes;
+  private long offsetBufferAllocationSizeInBytes;
+  private long validityBufferAllocationSizeInBytes;
+
+  private final FieldType fieldType;
+
+  private static final byte TYPE_WIDTH = 1;
+  public static final byte OFFSET_WIDTH = 4;
+
+  private static final FieldType INTERNAL_STRUCT_TYPE = new FieldType(false /*nullable*/,
```

Comment:
nit: lets use /*nullable=*/false, instead of putting it after the values.

Reply:
Revised. Thank you.